apps/nextjs-website/react-components/components/Accordion/AccordionItem.tsx

```diff
@@ -32,14 +39,20 @@ export const AccordionItem: React.FC<AccordionItemProps> = ({
         targetItem.scrollIntoView({ behavior: 'smooth', block: 'center' });
       }
     }
-  }, [])
+  }, []);
+
+  const handleChange = () => {
+    if (!expanded) {
+      appendItemIDToURLHash();
+    }
+    setExpanded(!expanded);
```

Comment:
```suggestion
    setExpanded(!expanded);
    if (!expanded) {
      appendItemIDToURLHash();
    }
```
Set Accordion state first, so it works even if `appendItemIDToURLHash` fails